DAOS-15695 ci: golang update to 1.22.3

grom72 opened the pull request:
Update Golang due to detected vulnerability in the previous versions (1.21.6)

grom72 committed:
DAOS-15695: golang update to 1.22.3

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
packaging update

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
changelog update

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

brianjmurrell commented on go.spec:
```diff
@@ -111,6 +111,9 @@ fi
 %endif
 
 %changelog
+* Thu Jan 18 2024 Tomasz Gromadzki <tomasz.gromadzki@intel.com> - 1.22.3-1
```
Need to update this datestamp.

grom72 replied:
Done

grom72 committed:
Update the date

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Update the date (day of week)

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Add files required for 'Check Packaging' step

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

brianjmurrell commented:
Hrm.  The [downstream testing](https://jenkins-1.wolf.hpdd.intel.com:9090/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-go-build-toolchain-PR-25-master/7/pipeline/182) is not using this PR's packages in it's RPM build setup.  I suspect that's because it's being set with the `CI_PR_REPOS` and not the `PR-repos:` commit pragma.  We are probably only checking and using the latter.

Probably/perhaps https://github.com/daos-stack/packaging/blob/1567c66dd52acdab63e50009115f2acde9212bef/Makefile_packaging.mk#L38 needs to check for and prefer the `CI_PR_REPOS` environment variable if it exists.  Could you try patching the `packaging/` subdir in this pr with:

```diff
--- a/Makefile_packaging.mk
+++ b/Makefile_packaging.mk
@@ -35,7 +35,11 @@ TEST_PACKAGES ?= ${NAME}
 # unfortunately we cannot always name the repo the same as the project
 REPO_NAME ?= $(NAME)
 
+ifneq ($(CI_PR_REPOS),)
+PR_REPOS                 ?= $(CI_PR_REPOS)
+else
 PR_REPOS                 ?= $(shell git show -s --format=%B | sed -ne 's/^PR-repos: *\(.*\)/\1/p')
+endif
 LEAP_15_PR_REPOS         ?= $(shell git show -s --format=%B | sed -ne 's/^PR-repos-leap15: *\(.*\)/\1/p')
 EL_7_PR_REPOS            ?= $(shell git show -s --format=%B | sed -ne 's/^PR-repos-el7: *\(.*\)/\1/p')
 EL_8_PR_REPOS            ?= $(shell git show -s --format=%B | sed -ne 's/^PR-repos-el8: *\(.*\)/\1/p')
```

on the hope that `CI_PR_REPOS` is exposed to the environment that `make` is running in?

If that doesn't work we might have to dig a bit deeper and see if that environment variable is set in the environment this script is running in: https://github.com/daos-stack/pipeline-lib/blob/31dfdfe7ae64715e2da0fadffaa0a8c3a4be3378/vars/buildRpm.groovy#L73-L74.  But let's just start with the patch above.  You may want to create a new PR based on this PR to do that testing in so as not to muck up this PR.

brianjmurrell reviewed (changes requested):
LGTM, but I'm just going to -1 until we work out the downstream testing issue.

grom72 committed:
Use CI_PR_REPOS in Dependency Validation Build Test if set

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

brianjmurrell commented:
@grom72 My apologies.  I wasn't considering where `CI_PR_REPOS` needs to be evaluated.  That needs to happen in the daos repo, not this one.

Can you create a daos PR and apply the below patch to it in addition to it being applied here?

> ```diff
> --- a/Makefile_packaging.mk
> +++ b/Makefile_packaging.mk
> @@ -35,7 +35,11 @@ TEST_PACKAGES ?= ${NAME}
>  # unfortunately we cannot always name the repo the same as the project
>  REPO_NAME ?= $(NAME)
>  
> +ifneq ($(CI_PR_REPOS),)
> +PR_REPOS                 ?= $(CI_PR_REPOS)
> +else
>  PR_REPOS                 ?= $(shell git show -s --format=%B | sed -ne 's/^PR-repos: *\(.*\)/\1/p')
> +endif
>  LEAP_15_PR_REPOS         ?= $(shell git show -s --format=%B | sed -ne 's/^PR-repos-leap15: *\(.*\)/\1/p')
>  EL_7_PR_REPOS            ?= $(shell git show -s --format=%B | sed -ne 's/^PR-repos-el7: *\(.*\)/\1/p')
>  EL_8_PR_REPOS            ?= $(shell git show -s --format=%B | sed -ne 's/^PR-repos-el8: *\(.*\)/\1/p')
> ```

Once you have that daos PR, you can add a new (empty, even) commit to this PR with the following in the commit message:

```
Test-master-branch: <daos-PR's/branch>
```

Note that `<daos-PR's/branch>` is the branch name (i.e. `grom72/use-CI_PR_REPOS`), not a PR-####.  You can push the updated commit to this PR as soon as you push the daos PR.  You don't need to wait for that PR to complete (or even be in) CI.

grom72 committed:
Force custom branch for validation

Test-master-branch: grom72/use-CI_PR_REPOS

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Retest.

Test-master-branch: grom72/use-CI_PR_REPOS

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Do not build packages in Dependency Validation Build Test

Wait for previous build to finish
Cancel-prev-build: false

Priority: 2

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Disable test_dfuse_daos_build_wt_pil4dfs test

https://daosio.atlassian.net/browse/DAOS-15834

Test-tag: daosio, -test_dfuse_daos_build_wt_pil4dfs

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 commented:
DAOS pipeline adjusted in https://github.com/daos-stack/daos/pull/14387

grom72 committed:
Revert "Do not build packages in Dependency Validation Build Test"

This reverts commit c6a08c7c0b6e30d2d9542ca8f6b0e886a4446dfd.

Priority: 2

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Disable test_dfuse_daos_build_wt_pil4dfs test

Skip-list: test_dfuse_daos_build_wt_pil4dfs:DAOS-15384
Priority: 2

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Final test for grom72/use-CI_PR_REPOS

Test-master-branch: grom72/use-CI_PR_REPOS

Skip-list: test_dfuse_daos_build_wt_pil4dfs:DAOS-15384
Priority: 2

Skip-build-ubuntu20-rpm: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Enable Ubuntu build

Test-master-branch: grom72/use-CI_PR_REPOS

Skip-list: test_dfuse_daos_build_wt_pil4dfs:DAOS-15384
Priority: 2

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Revert "Enable Ubuntu build"

This reverts commit fae966f92e61be96e654b5090428384395222cfb.

Skip-list: test_dfuse_daos_build_wt_pil4dfs:DAOS-15384
Priority: 2

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Skip Ubuntu build

Skip-build-ubuntu20-rpm: true
Skip-list: test_dfuse_daos_build_wt_pil4dfs:DAOS-15384
Priority: 2

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Skip ubuntu build (proper test branch)

Skip-build-ubuntu20-rpm: true

Test-master-branch: grom72/use-CI_PR_REPOS

Skip-list: test_dfuse_daos_build_wt_pil4dfs:DAOS-15384
Priority: 2

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Verify new way to ignore Ubuntu in downstream tests

Test-master-branch: grom72/use-CI_PR_REPOS

Skip-list: test_dfuse_daos_build_wt_pil4dfs:DAOS-15384
Priority: 2

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Merge latest packaging w/ HPE repo key from Artifactory

Test-master-branch: grom72/use-CI_PR_REPOS

Skip-list: test_dfuse_daos_build_wt_pil4dfs:DAOS-15384
Priority: 2

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Revert "Verify new way to ignore Ubuntu in downstream tests"

This reverts commit 1197520bce5e24eaf53f8466e8002dfec81dcf23.

Test-master-branch: grom72/use-CI_PR_REPOS

Skip-list: test_dfuse_daos_build_wt_pil4dfs:DAOS-15384
Priority: 2

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

brianjmurrell committed:
Update packaging/

Priority: 2
Required-githooks: true

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

grom72 committed:
Validate with suppressed new Valgrind issues

Test-master-branch: grom72/DAOS-16011
Skip-func-hw-test: true

Priority: 2
Required-githooks: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com

grom72 committed:
Validate with suppressed new Valgrind issues (2nd)

Test-master-branch: grom72/DAOS-16011
Skip-func-hw-test: true

Priority: 2
Required-githooks: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com

grom72 committed:
Revert "Validate with suppressed new Valgrind issues (2nd)"

This reverts commit 029897caf766e42b57e0d704218a21981e212026.

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Revert "Validate with suppressed new Valgrind issues"

This reverts commit 17e22a4bf29281f2f3b0ad36bb5bc5a0bd5c90b1.

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Prepare for final validation

Skip HW tests as they are not related to go update.
Skip-func-hw-test: true

Priority: 2
Required-githooks: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
Final validation

Priority: 2
Required-githooks: true

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

ryon-jensen committed:
Validation with valgrind supression

Skip-func-test: true

Signed-off-by: Ryon Jensen <ryon.jensen@intel.com>

ryon-jensen commented:
[Build 23](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-go-build-toolchain-PR-25-release-2.6/23/pipeline/) had a valgrind error in NLT. @grom72 created a fix (suppression file). Then [build 26](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-go-build-toolchain-PR-25-release-2.6/26/pipeline) with the fix applied passed NLT but had a failure in HW tests. These failures look unrelated to a go package update. I think this should be okay to land now.

grom72 commented:
All HW tests passed in [Build 23](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-go-build-toolchain-PR-25-release-2.6/23/pipeline) (No changes between build 23 and 26)
